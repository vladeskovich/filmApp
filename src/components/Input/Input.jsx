import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Input.scss';

const Input = ({
  type,
  className,
  onChange,
  ...props
}) => {
  const classes = classNames(styles.input, className, type);

  return (
    <input
      type={type}
      className={classes}
      onChange={onChange}
      {...props}
    />
  );
};

Input.propTypes = {
  type: PropTypes.oneOf(['checkbox', 'radio', 'submit', 'text']),
  className: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
};

Input.defaultProps = {
  type: 'text',
};

export default Input;
