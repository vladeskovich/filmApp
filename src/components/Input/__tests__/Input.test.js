import React from 'react';
import Input from '../Input';

describe('Input', () => {
  const props = {
    type: 'radio',
    className: 'inputTest',
    handleChange: () => {},
  };

  it('should render Input component with props', () => {
    const component = shallow(<Input {...props}/>);

    expect(component).toMatchSnapshot();
  });

  it('should call onChange method', () => {
    const mockCallBack = jest.fn();
    const event = {
      preventDefault() {},
      target: { value: 'test-value' },
    };

    const component = shallow(<Input onChange={mockCallBack}/>);

    component.find('input').simulate('change', event);

    expect(mockCallBack).toBeCalledWith(event);
  });
});
