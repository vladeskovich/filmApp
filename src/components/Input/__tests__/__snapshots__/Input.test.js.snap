// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input component with props 1`] = `
<input
  className="Input__input_1c_Lh1r3 inputTest radio"
  handleChange={[Function]}
  type="radio"
/>
`;
